crates/flowctl/src/raw/oauth.rs

```diff
@@ -0,0 +1,273 @@
+    source: String,
+    /// Name of the task to test oauth for.
+    #[clap(long)]
+    task: Option<String>,
+    /// Docker network to run the connector, if one exists
+    #[clap(long, default_value = "bridge")]
+    network: String,
+
+    /// Your OAuth application's client ID. This corresponds to the
+    /// `connectors.oauth2_client_id` column in the database
+    #[clap(long)]
+    client_id: String,
+    /// Your OAuth application's client secret. This corresponds to the
+    /// `connectors.oauth2_client_secret` column in the database
+    #[clap(long)]
+    client_secret: String,
+    /// The endpoint config you want to test with. This is the config for a
+    /// sample user account that will perform the end-to-end consent flow
+    #[clap(long)]
+    endpoint_config: serde_json::Value,
+    /// If the connector's oauth spec required needs any other values injected.
+    /// This will match the `connectors.oauth2_injected_values` column in the database.
+    #[clap(long)]
+    injected_values: Option<serde_json::Value>,
+}
+
+pub async fn do_oauth(
+    ctx: &mut crate::CliContext,
+    Oauth {
+        source,
+        task,
+        network,
+        client_id,
+        client_secret,
+        endpoint_config,
+        injected_values,
+    }: &Oauth,
+) -> anyhow::Result<()> {
+    // TESTING
+    let env_filter = tracing_subscriber::EnvFilter::builder()
+        .with_default_directive(tracing_subscriber::filter::LevelFilter::INFO.into()) // Otherwise it's ERROR.
+        .from_env_lossy();
+
+    let guard = tracing::subscriber::set_default(
+        tracing_subscriber::fmt::fmt()
+            .with_env_filter(env_filter)
+            .compact()
+            .without_time()
+            .with_target(false)
+            .with_writer(std::io::stderr)
+            .finish(),
+    );
+    // END TESTING
+    let source = build::arg_source_to_url(source, false)?;
+    let mut sources = local_specs::surface_errors(local_specs::load(&source).await.into_result())?;
+
+    // Identify the capture to inspect.
+    // TODO (jshearer): materialization support
+    let needle = if let Some(needle) = task {
+        needle.as_str()
+    } else if sources.captures.len() == 1 {
+        sources.captures.first().unwrap().capture.as_str()
+    } else if sources.captures.is_empty() {
+        anyhow::bail!("sourced specification files do not contain any captures");
+    } else {
+        anyhow::bail!("sourced specification files contain multiple captures. Use --capture to identify a specific one");
+    };
+
+    let capture = match sources
+        .captures
+        .binary_search_by_key(&needle, |c| c.capture.as_str())
+    {
+        Ok(index) => &mut sources.captures[index],
+        Err(_) => anyhow::bail!("could not find the capture {needle}"),
+    };
+
+    tracing::info!(
+        task_name = capture.capture.as_str(),
+        "Performing oauth flow on task"
+    );
+
+    let spec_req = match &capture.spec.endpoint {
```

Comment:
No action for now, but I'll point out that we'd like to eventually switch the `connector_tags` job over to using the `runtime` crate to pull connector specs like this. I imagine this section could be easily factored out and exposed as a `async fn spec(...)`, which could also be used by `agent`.